code/parachain/frame/composable-tests-helpers/src/test/helper.rs

```diff
@@ -41,186 +41,187 @@ pub fn default_acceptable_computation_error(x: u128, y: u128) -> Result<(), Fixe
+pub trait RuntimeTrait<PalletEvent>: Config
 where
-	Runtime: Config<Event = RuntimeEvent>,
-	PalletEvent: sp_std::fmt::Debug + Clone,
-	RuntimeEvent: TryInto<PalletEvent> + Parameter + Member + Debug + Clone,
-	<RuntimeEvent as TryInto<PalletEvent>>::Error: sp_std::fmt::Debug,
+	<Self as frame_system::Config>::Event:
+		Parameter + Member + Debug + Clone + TryInto<PalletEvent> + From<PalletEvent>,
+	<<Self as frame_system::Config>::Event as TryInto<PalletEvent>>::Error: Debug,
+	<Self as frame_system::Config>::Origin:
+		OriginTrait<AccountId = <Self as frame_system::Config>::AccountId>,
+	PalletEvent: Clone + Debug + PartialEq,
 {
-	// compare to the last event record
-	let EventRecord { event, .. } =
-		frame_system::Pallet::<Runtime>::events().pop().expect("No events present!");
-
-	match event.clone().try_into() {
-		Ok(pallet_event) => match f(pallet_event.clone()) {
-			Some(r) => r,
-			None => panic!("expected event was not found; found {pallet_event:#?}"),
-		},
-		Err(_) => panic!(
-			r#"last event was not from this pallet
+	/// Asserts that the last event in the runtime is the expected event.
+	fn assert_last_event(pallet_event: PalletEvent) {
+		let events = frame_system::Pallet::<Self>::events();
+		// compare to the last event record
+		let EventRecord { event, .. } = &events.last().expect("No events present!");
+		assert_eq!(event, &pallet_event.into());
+	}
+
+	/// Asserts that the last event in the runtime is the expected event.
+	///
+	/// Useful if not all of the information in the event needs to be checked:
+	///
+	/// ```rust,ignore
+	/// assert_last_event_with::<Runtime, _>(
+	///     Pallet::extrinsic(),
+	///     |event| matches!(
+	///         event,
+	///         pallet::Event::<Runtime>::SomethingHappened {
+	///             field,
+	///             ..
+	///         } if field == expected_field
+	///     ).then_some(())
+	/// )
+	/// ```
+	///
+	/// It is also possible to return a value from the provided function, for example to retrieve a
+	/// generated id for later use:
+	///
+	/// ```rust,ignore
+	/// assert_last_event_with::<Runtime, _>(
+	///     Pallet::extrinsic(),
+	///     |event| if let pallet::Event::<Runtime>::SomethingHappened {
+	///         field,
+	///         generated_id,
+	///     } = event {
+	///         assert!(field);
+	///         Some(generated_id)
+	///     } else {
+	///         None
+	///     },
+	/// )
+	/// ```
+	fn assert_last_event_with<R>(f: impl FnOnce(PalletEvent) -> Option<R>) -> R {
+		// compare to the last event record
+		let EventRecord { event, .. } =
+			frame_system::Pallet::<Self>::events().pop().expect("No events present!");
+
+		match event.clone().try_into() {
+			Ok(pallet_event) => match f(pallet_event.clone()) {
+				Some(r) => r,
+				None => panic!("expected event was not found; found {pallet_event:#?}"),
+			},
+			Err(_) => panic!(
+				r#"last event was not from this pallet
 found {event:#?}"#
```

Comment:
NIT
Indenting on this looks funky
(and on other uses of `r#".."#)

Reply:
With raw strings, leading spaces are included in the string itself, so there's not much I can do about it other than make them all start with a newline like this:

```rust
let s = r#"
Blah
Blah
Blah
"#;

(Which I actually prefer and try do do, I'll update any that aren't formatted that way 👍)

Reply:
I think I've fixed all of these, lmk if I missed any!